[IC2_BE] 유효진- 설문조사 서비스 과제

youabledev opened the pull request:
# DB 정보
## ERD
![스크린샷 2024-11-30 오전 10 34 57](https://github.com/user-attachments/assets/af1014af-c432-4b20-807f-c8d619c6b8d0)

# API 명세
https://github.com/youabledev/be-onboarding-project/blob/main/README.md

# 진행사항
## 기본사항
- [x] 설문조사 생성 API
    - [ ] 테스트 코드
    - [x] 명세
- [x] 설문조사 수정 API
    - [ ] 테스트 코드
    - [x] 명세
- [x] 설문조사 응답 제출 API
    - [ ] 테스트 코드
    - [x] 명세
- [x] 설문조사 응답 조회 API 
    - [ ] 테스트 코드
    - [x] 명세
## Advanced 
- [ ] 설문 응답 항목의 이름과 응답 값 기반의 검색
## 우대사항
- [ ] 트래픽이 많고, 저장되어 있는 데이터가 많음을 염두에 둔 구현
- [ ] 다수의 서버, 인스턴스에서 동작할 수 있음을 염두에 둔 구현
- [ ] 동시성 이슈가 발생할 수 있는 부분을 염두에 둔 구현

## 궁금한 것 
- 처음에 설문 받을 항목에 포함되는 후보 값을 구분자를 포함한 하나의 문자열로 받으려고 했습니다 (ex "옵션1, 옵션2, 옵션3")
그러다 후보 값 자체에 구분자가 들어갈 수도 있을 것 같아 option이라는 테이블을 생성했습니다. 설문조사 수정 구현시 꽤나 복잡해져서 이렇게 하는게 좋을지, 후보값에 들어가지 않을 구분자를 사용해 하나의 문자열로 저장하면 좋을지 고민이 되었습니다.

Commit:
add .gitigonore

Commit:
project initialize

youabledev commented:
Labels 설정이 되지 않습니다!

heli-os commented:
<img width="288" alt="Screenshot 2024-11-24 at 23 52 41" src="https://github.com/user-attachments/assets/188bca83-9487-457e-8697-3c72a2c0f52a">

@youabledev 여기서 안되실까요? 권한 이슈인거 같기도한데, 확인해보겠습니다~

youabledev commented:
네, 우측의 톱니바퀴 아이콘이 보이지 않아요!

heli-os commented:
> @heli-os 네, 우측의 톱니바퀴 아이콘이 보이지 않아요!

다시 한번 확인 해주실 수 있을까요?

youabledev commented:
labels 추가 잘 됩니다! 감사합니다~

Commit:
Refactor: application.properties -> application.yml

Commit:
Feat: h2 dbms 설정

Commit:
Feat: JPA Entity 추가
- 설문 생성 및 관리를 위한 Survey, Question, Option 엔티티 구현
- 설문 응답 저장을 위한 Response, Answer 엔티티 구현

youabledev committed:
Add: QueryDSL dependency 추가

youabledev committed:
Add: 전역으로 EntityManager 사용하기 위해 Config 추가

youabledev committed:
Add: Survey Entity의 Repository 추가

youabledev committed:
Fix: 누락된 연관관계 수정

youabledev committed:
Refactor: Entity의 idx는 각 클래스에서 관리하도록 수정

youabledev committed:
Feat: 설문조사 생성 API

Commit:
Fix: gitignore 누락 사항 추가

youabledev committed:
Refactor: 설문조사 생성 API 개선

youabledev committed:
Add: RegistSurveyResponse 추가

youabledev committed:
Add: 공통 Response 추가

youabledev committed:
Add: Valid dependency 추가

youabledev committed:
Feat: valid 검증을 위한 ExceptionHandler 로직 추가

youabledev committed:
Feat: SruveyRequest 검증

youabledev committed:
Feat: 설문조사 생성 요청 값 검증 추가

youabledev committed:
Add: 비즈니스 로직 에러시 사용할 ServiceException 추가

youabledev committed:
Refactor: 설문 조사 조회시 질문과 옵션을 한번에 가져오기 위해 Entity 필드 추가

youabledev committed:
Revert "Refactor: 설문 조사 조회시 질문과 옵션을 한번에 가져오기 위해 Entity 필드 추가"

This reverts commit a62224129f2846deff96cf892ef1a29ff14158c7.

youabledev committed:
Feat: 설문 응답 제출 API

youabledev committed:
Refactor: Option, Question관련 JpaRepository에서 QueryDSL로 변경

youabledev committed:
Refactor: Survey관련 JpaRepository에서 QueryDSL로 변경

youabledev committed:
Fix: 설문 응답 생성시 영속성 이슈 수정

youabledev committed:
Refactor: 설문 조회 시 설문 응답 목록도 함께 가져오도록 변경

youabledev committed:
Feat: CreateAnswerRequest validation 추가

youabledev committed:
Refactor: 설문 응답 제출시 응답 데이터

youabledev committed:
Refactor: 설문 조사 생성 RegistSurveyResponse 객체 개선

youabledev committed:
Feat: 설문조사 응답 제출 API response

Commit:
Merge branch 'main' of github.com:youabledev/be-onboarding-project into main

Commit:
Refactor: 설문조사 Request 객체 변경
- @Builder가 아닌 toEntity 메소드 사용해서 Request에서 Entity로 변환 가능하도록 변경

Commit:
Fix: Options 데이터 없는 경우 발생하는 문제 수정

Commit:
Fix: Options 데이터 없는 경우 발생하는 문제 수정

Commit:
Feat: 설문조사 수정 api

Commit:
Feat: 설문조사 생성, 설문조사 수정 API에 대한 API 명세서

Commit:
Fix: 옵션 이슈 수정

youabledev committed:
Refactor: 설문조사 생성 API 코드 개선

youabledev committed:
Refactor: 설문조사 응답 제출 API 개선

youabledev committed:
Refactor: 설문 등록시 확인해야 하는 Option idx 값 추가

youabledev committed:
Fix: 설문조사 응답 조회 API 이슈 수정

Commit:
Refactor: 설문조사 수정 API 개선

youabledev commented:
@heli-os 
아직 테스트 코드도 못 작성했고 리뷰 요청도 늦은 것 같지만 혹시 코드 리뷰 받을 수 있을까요?? 🥲
DB 설계와 Entity 클래스의 메소드 구현이 올바른 방향인지 궁금합니다.
많이 부족한 실력이지만, 조금이라도 개선할 수 있도록 코드 리뷰를 해주신다면 정말 감사하겠습니다. 🥹

Commit:
Refactor: SurveyService 불필요한 코드 정리

Commit:
Add: Valid 테스트 코드 추가

Commit:
Add: repository code 작업 중

youabledev committed:
Fix: 유효하지 않은 QuestionType이 요청된 경우 발생하는 에러 처리

youabledev committed:
Feat: Survey Request DTO Valid Test
- QuestionRequestTest
- SurveyRequetTest

youabledev committed:
Merge remote-tracking branch 'origin/main'

youabledev committed:
Feat: Survey 도메인의 Repository 테스트 추가

youabledev committed:
Feat: Service Repository 테스트 코드

youabledev committed:
Feat: SurveyService test

youabledev committed:
Feat: SubmissionCreationRequest test

youabledev committed:
Refactor: 설문조사 추가, 업데이트 리팩토링
